### ISSUE TITLE
Reopening issue - Unable to fine tune en-indic model #33

### ISSUE THREAD

**Issue by PS-AI** [AI4Bharat/IndicTrans2#39]
Hi,
My query is still pending, I am unable to finetune the en-indic model. I have given the details in my respose to @jaygala24 in Issue #33.

Request for resolution please.

**Comment by PS-AI** [comment]
Hi @prajdabre,

Thank you for your work.

This is the finetune.sh script that I am using:

#!/bin/bash

# This script finetunes the pretrained translation model on the binarized data using fairseq.


echo `date`
exp_dir=$1                              # path of the experiment directory
model_arch=${2:-"transformer_18_18"}    # model architecture (defaults to `transformer_18_18`)
pretrained_ckpt=$3                      # path to the pretrained checkpoint `.pt` file


fairseq-train $exp_dir/final_bin \
--max-source-positions=256 \
--max-target-positions=256 \
--source-lang=SRC \
--target-lang=TGT \
--max-update=25 \
--save-interval-updates=10 \
--arch=$model_arch \
--activation-fn gelu \
--criterion=label_smoothed_cross_entropy \
--label-smoothing=0.1 \
--optimizer adam \
--adam-betas "(0.9, 0.98)" \
--lr-scheduler=inverse_sqrt \
--clip-norm 1.0 \
--warmup-init-lr 1e-07 \
--lr 3e-5 \
--warmup-updates 2000 \
--dropout 0.2 \
--save-dir $exp_dir/model \
--keep-last-epochs 5 \
--keep-interval-updates 3 \
--patience 10 \
--skip-invalid-size-inputs-valid-test \
--fp16 \
--user-dir model_configs \
--update-freq=4 \
--distributed-world-size 1 \
--num-workers 1 \
--max-tokens 1024 \
--eval-bleu \
--eval-bleu-args "{\"beam\": 1, \"lenpen\": 1.0, \"max_len_a\": 1.2, \"max_len_b\": 10}" \
--eval-bleu-detok moses \
--eval-bleu-remove-bpe sentencepiece \
--eval-bleu-print-samples \
--best-checkpoint-metric bleu \
--maximize-best-checkpoint-metric \
--task translation \
--finetune-from-model $pretrained_ckpt

In the command line argument for the finetune.sh( 'path to the pretrained checkpoint `.pt` file') - I am providing the path to the downloaded fairseq model downloaded using this link:
https://indictrans2-public.objectstore.e2enetworks.net/it2_preprint_ckpts/en-indic-preprint.zip

When I run finetune.sh, training starts from epoch1 instead of from the last epoch that the downloaded model has been trained for.  This is the issue that I am facing.

**Comment by jaygala24** [comment]
The IndicTrans2 models publicly released as a part of this work are fine-tuned on human-annotated seed data composed of BPCC-H-Wiki and NLLB-Seed. These models typically converge within a single epoch on the provided seed data. As a result, the scheduler starts from epoch 1 which is indeed the last epoch for these models.
